test/server/cards/02_SHD/leaders/BobaFettDaimyo.spec.ts

```diff
@@ -0,0 +1,141 @@
+                expect(context.player1).toBeAbleToSelectExactly([context.battlefieldMarine, context.wildernessFighter, context.greenSquadronAwing, context.cantinaBraggart]);
+
+                // boost battlefield marine
+                context.player1.clickCard(context.battlefieldMarine);
+                expect(context.bobaFett.exhausted).toBeTrue();
+                expect(context.battlefieldMarine.getPower()).toBe(4);
+                expect(context.battlefieldMarine.getHp()).toBe(3);
+            });
+        });
+
+        describe('Boba Fett\'s leader ability', function () {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['alliance-xwing'],
+                        groundArena: ['battlefield-marine'],
+                        spaceArena: ['red-three#unstoppable'],
+                        leader: 'boba-fett#daimyo',
+                        resource: 4
+                    },
+                    player2: {}
+                });
+            });
+
+            it('should give +1/+0 to other friendly unit when play a unit which gain a keyword by an ability', function () {
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.allianceXwing);
+                // boba triggers as red three give raid 1 to alliance x-wing
+                expect(context.player1).toHavePrompt('Choose a friendly to give +1/+0 for this phase');
+                expect(context.player1).toHavePassAbilityButton();
+                expect(context.player1).toBeAbleToSelectExactly([context.battlefieldMarine, context.redThree, context.allianceXwing]);
+
+                // give +1/+0 to battlefield marine, boba should be exhausted
+                context.player1.clickCard(context.battlefieldMarine);
+                expect(context.bobaFett.exhausted).toBeTrue();
+                expect(context.battlefieldMarine.getPower()).toBe(4);
+                expect(context.battlefieldMarine.getHp()).toBe(3);
+            });
+        });
+
+        // TODO should add tests with timely intervention or ecl
+
+        describe('Boba Fett\'s leader unit ability', function () {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['red-three#unstoppable'],
+                        groundArena: ['battlefield-marine'],
+                        spaceArena: ['green-squadron-awing'],
+                        leader: { card: 'boba-fett#daimyo', deployed: true },
+                    },
+                    player2: {
+                        spaceArena: ['outer-rim-headhunter'],
+                    }
+                });
+            });
+
+            it('should give +1/+0 to other friendly unit with keyword', function () {
```

Comment:
Just to check that the engine behavior is right, could you add an expectation that cards in hand do _not_ have the +1 if they have a keyword? That way if we ever mess with the targeting for ongoing effects we'll know if we break something

Reply:
skip as we can't check power modification on hand